pennylane/devices/default_clifford.py

```diff
@@ -102,28 +103,53 @@
     "StatePrep": None,
     "Snapshot": None,
     "Barrier": None,
+    "PauliError": "CORRELATED_ERROR",
+    "BitFlip": "X_ERROR",
+    "PhaseFlip": "Z_ERROR",
+    "DepolarizingChannel": "DEPOLARIZE1",
 }
 
 
 def operation_stopping_condition(op: qml.operation.Operator) -> bool:
     """Specifies whether an operation is accepted by ``DefaultClifford``."""
-    return op.name in _GATE_OPERATIONS
+    return op.name in _OPERATIONS_MAP
 
 
 def observable_stopping_condition(obs: qml.operation.Operator) -> bool:
     """Specifies whether an observable is accepted by ``DefaultClifford``."""
-    return obs.name in _MEAS_OBSERVABLES
+    return obs.name in _OBSERVABLES_MAP
+
+
+@qml.transform
+def _validate_channels(tape, stopping_condition, name="device"):
+    """Validates the channels for a circuit."""
+    for op in tape.operations:
+        if isinstance(op, qml.operation.Channel):
+            if not stopping_condition(op):
+                raise qml.DeviceError(f"Channel {repr(op)} not supported on {name}")
```

Comment:
I feel a little funny about this. `decompose` runs before with the same stopping condition, so we should theoretically never hit this line (these should either be decomposed or it should have failed). It can only happen if `check_clifford=False`, which is already being sneaky and letting the user override some things.

After typing this, I think my proposal is either:
1. remove `check_clifford` (I think we talked enough about this, and we're keeping it, soooo option 2 might be best)
2. only validate channels if `check_clifford` is True, and remove the `stopping_condition` from this transform. If the user wants to skip it, let them skip it!